### DISCUSSION TITLE
Open script in editor automatically after creation via the Create Script dialog box

### DISCUSSION THREAD

**Discussion by pahool** [godotengine/godot-proposals#11428] [Editor]
In the filesystem dock, when creating a new script via right-clicking and selecting Create New -> Script. A "Create Script" dialog box is opened. Currently, after creating a script, the file is not opened automatically in the editor. So the created script must be opened by navigating the filesystem dock and opening the newly-created script. I would submit that most users, when creating a script, want to begin editing that script and so the default behavior should be to open the newly-created script in the editor. This default behavior could be toggled on/off in project settings and/or via a checkbox in the "Create Script" dialog box itself.

**Comment by pahool** [comment]
I went ahead and created an addon that does this. I'm going to test and enhance it a bit before pushing to the asset library. You can check it out here and I'd appreciate constructive feedback:

https://gitlab.com/pahool_gamedev/godot_file_autoopen_addon

I still think this should be the default behavior in Godot

**Comment by JSalske** [comment]
I couldn't agree more. This would be a big help to my personal workflow.

**Comment by timothyqiu** [comment]
Implemented in https://github.com/godotengine/godot/pull/108342

**Reply by pahool** [reply]
Excellent. Hope this gets merged. A much better solution than a GDscript-based addon.

**Reply by pahool** [reply]
Looks like it's on-track to be in 4.6. Excellent!
